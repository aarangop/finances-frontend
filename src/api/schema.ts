/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */

export interface paths {
    "/vehicles/": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Read Vehicles */
        get: operations["read_vehicles_vehicles__get"];
        put?: never;
        /** Create Vehicle */
        post: operations["create_vehicle_vehicles__post"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
    "/vehicles/{vehicle_id}": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Read Vehicle */
        get: operations["read_vehicle_vehicles__vehicle_id__get"];
        /** Update Vehicle */
        put: operations["update_vehicle_vehicles__vehicle_id__put"];
        post?: never;
        /** Delete Vehicle */
        delete: operations["delete_vehicle_vehicles__vehicle_id__delete"];
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
    "/trips/": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Read Trips */
        get: operations["read_trips_trips__get"];
        put?: never;
        /** Create Trip */
        post: operations["create_trip_trips__post"];
        delete?: never;
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
    "/trips/{trip_id}": {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        /** Read Trip */
        get: operations["read_trip_trips__trip_id__get"];
        /** Update Trip */
        put: operations["update_trip_trips__trip_id__put"];
        post?: never;
        /** Delete Trip */
        delete: operations["delete_trip_trips__trip_id__delete"];
        options?: never;
        head?: never;
        patch?: never;
        trace?: never;
    };
}
export type webhooks = Record<string, never>;
export interface components {
    schemas: {
        /** HTTPValidationError */
        HTTPValidationError: {
            /** Detail */
            detail?: components["schemas"]["ValidationError"][];
        };
        /** Trip */
        Trip: {
            /** From Location */
            from_location: string;
            /** To Location */
            to_location: string;
            /**
             * Start Date
             * Format: date
             */
            start_date: string;
            /**
             * End Date
             * Format: date
             */
            end_date: string;
            /** Odometer Start */
            odometer_start: number;
            /** Odometer End */
            odometer_end: number;
            /** Vehicle Id */
            vehicle_id: number;
            /** Id */
            id: number;
        };
        /** TripCreate */
        TripCreate: {
            /** From Location */
            from_location: string;
            /** To Location */
            to_location: string;
            /**
             * Start Date
             * Format: date
             */
            start_date: string;
            /**
             * End Date
             * Format: date
             */
            end_date: string;
            /** Odometer Start */
            odometer_start: number;
            /** Odometer End */
            odometer_end: number;
            /** Vehicle Id */
            vehicle_id: number;
        };
        /** ValidationError */
        ValidationError: {
            /** Location */
            loc: (string | number)[];
            /** Message */
            msg: string;
            /** Error Type */
            type: string;
        };
        /** Vehicle */
        Vehicle: {
            /** Name */
            name: string;
            /** Brand */
            brand: string;
            /** Model */
            model: string;
            /** Year */
            year: number;
            /** Id */
            id: number;
            /**
             * Trips
             * @default []
             */
            trips: components["schemas"]["Trip"][];
        };
        /** VehicleCreate */
        VehicleCreate: {
            /** Name */
            name: string;
            /** Brand */
            brand: string;
            /** Model */
            model: string;
            /** Year */
            year: number;
        };
    };
    responses: never;
    parameters: never;
    requestBodies: never;
    headers: never;
    pathItems: never;
}
export type $defs = Record<string, never>;
export interface operations {
    read_vehicles_vehicles__get: {
        parameters: {
            query?: {
                skip?: number;
                limit?: number;
            };
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Vehicle"][];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    create_vehicle_vehicles__post: {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody: {
            content: {
                "application/json": components["schemas"]["VehicleCreate"];
            };
        };
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Vehicle"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    read_vehicle_vehicles__vehicle_id__get: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                vehicle_id: number;
            };
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Vehicle"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    update_vehicle_vehicles__vehicle_id__put: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                vehicle_id: number;
            };
            cookie?: never;
        };
        requestBody: {
            content: {
                "application/json": components["schemas"]["VehicleCreate"];
            };
        };
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Vehicle"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    delete_vehicle_vehicles__vehicle_id__delete: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                vehicle_id: number;
            };
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            204: {
                headers: {
                    [name: string]: unknown;
                };
                content?: never;
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    read_trips_trips__get: {
        parameters: {
            query?: {
                skip?: number;
                limit?: number;
            };
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Trip"][];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    create_trip_trips__post: {
        parameters: {
            query?: never;
            header?: never;
            path?: never;
            cookie?: never;
        };
        requestBody: {
            content: {
                "application/json": components["schemas"]["TripCreate"];
            };
        };
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Trip"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    read_trip_trips__trip_id__get: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                trip_id: number;
            };
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Trip"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    update_trip_trips__trip_id__put: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                trip_id: number;
            };
            cookie?: never;
        };
        requestBody: {
            content: {
                "application/json": components["schemas"]["TripCreate"];
            };
        };
        responses: {
            /** @description Successful Response */
            200: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["Trip"];
                };
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
    delete_trip_trips__trip_id__delete: {
        parameters: {
            query?: never;
            header?: never;
            path: {
                trip_id: number;
            };
            cookie?: never;
        };
        requestBody?: never;
        responses: {
            /** @description Successful Response */
            204: {
                headers: {
                    [name: string]: unknown;
                };
                content?: never;
            };
            /** @description Validation Error */
            422: {
                headers: {
                    [name: string]: unknown;
                };
                content: {
                    "application/json": components["schemas"]["HTTPValidationError"];
                };
            };
        };
    };
}
