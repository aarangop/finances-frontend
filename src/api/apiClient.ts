import createClient from "openapi-fetch";
import { paths } from "./schema";

const client = createClient<paths>({
  baseUrl: process.env.NEXT_PUBLIC_API_URL,
});

export default client;
