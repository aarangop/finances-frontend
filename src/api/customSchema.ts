import { components } from "./schema";

type BalanceUpdate = components["schemas"]["BalanceUpdateSchema"];
export type BalanceUpdateWithDate = BalanceUpdate & { date: Date };
