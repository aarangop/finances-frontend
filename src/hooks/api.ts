import OpenApiClientContext from "@/context/OpenApiClientContext";
import { useQueryClient } from "@tanstack/react-query";
import { useContext } from "react";

export const useApi = () => {
  return { openApiClient: useOpenApiClient(), queryClient: useQueryClient() };
};

export const useOpenApiClient = () => {
  let client = useContext(OpenApiClientContext);
  if (!client) {
    throw new Error(
      "useOpenApiClient must be used within an ApiClientsProvider"
    );
  }
  return client;
};
