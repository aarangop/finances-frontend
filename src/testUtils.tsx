import Providers from "@/components/providers/Providers";
import { render, RenderOptions } from "@testing-library/react";
import { AppRouterContext } from "next/dist/shared/lib/app-router-context.shared-runtime";
import { ReactElement } from "react";

const mockRouter = {
  push: jest.fn(),
  replace: jest.fn(),
  prefetch: jest.fn().mockResolvedValue(undefined),
  pathname: "/",
  route: "/",
  asPath: "/",
  query: {},
  isFallback: false,
  basePath: "",
  events: {
    on: jest.fn(),
    off: jest.fn(),
    emit: jest.fn(),
  },
  beforePopState: jest.fn(() => null),
  isReady: true,
  back: jest.fn(),
  forward: jest.fn(),
  refresh: jest.fn(),
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(
    <AppRouterContext.Provider value={mockRouter}>
      <Providers children={ui} />
    </AppRouterContext.Provider>
  );

// Re-export everything from @testing-library/react
export * from "@testing-library/react";

// Override render method with our custom one
export { customRender as render };
