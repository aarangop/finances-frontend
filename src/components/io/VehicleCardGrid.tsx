"use client";

import { components } from "@/api/schema";
import { Container, Grid, Typography } from "@mui/material";
import VehicleCard from "../cards/VehicleCard";
import VehicleCardSkeleton from "../cards/VehicleCardSkeleton";

type Vehicle = components["schemas"]["VehicleSchema"];

export interface VehicleCardGridProps {
  isLoading: boolean;
  isError: boolean;
  isSuccess: boolean;
  vehicles?: Vehicle[];
}
export default function VehicleCardGrid({
  isLoading,
  isError,
  isSuccess,
  vehicles = [],
}: VehicleCardGridProps) {
  return (
    <Container sx={{ display: "flex" }}>
      {isError && (
        <Typography variant="h6" color="error">
          Error loading vehicles
        </Typography>
      )}
      {!isError && (
        <Grid container spacing={2} sx={{}}>
          {isSuccess &&
            vehicles.map((vehicle) => (
              <Grid item key={vehicle.id} xs={2} sm={4} md={4}>
                <VehicleCard vehicle={vehicle} />
              </Grid>
            ))}
          {isLoading &&
            Array.from({ length: 2 }).map((_, index) => (
              <Grid item key={index} xs={2} sm={4} md={4}>
                <VehicleCardSkeleton />
              </Grid>
            ))}
        </Grid>
      )}
    </Container>
  );
}
