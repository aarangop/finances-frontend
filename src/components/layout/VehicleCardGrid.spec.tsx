import { components } from "@/api/schema";
import { render } from "@/testUtils";
import VehicleCardGrid from "./VehicleCardGrid";

type Vehicle = components["schemas"]["VehicleSchema"];
const mockVehicles: Vehicle[] = [
  {
    id: 1,
    name: "Car 1",
    make: "Toyota",
    model: "Corolla",
    year: 2020,
    odometer: 12950,
    color: "White",
    license_plate: "AAA-111",
    vehicle_type: "car",
  } as Vehicle,
  {
    id: 2,
    name: "Car 2",
    make: "BMW",
    model: "X3",
    year: 2014,
    odometer: 67909,
    color: "red",
    license_plate: "bbb-111",
    vehicle_type: "car",
  } as Vehicle,
  {
    id: 3,
    name: "Motorcycle 1",
    make: "Royal Enfield",
    model: "Himalayan 400",
    year: 2021,
    odometer: 10560,
    color: "Black",
    license_plate: "ccc-111",
    vehicle_type: "motorcycle",
  } as Vehicle,
];
describe("VehicleCardGrid", () => {
  it("should render the VehicleCardGrid component", () => {
    const { container } = render(
      <VehicleCardGrid
        isSuccess={true}
        isError={false}
        vehicles={mockVehicles}
        isLoading={false}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
