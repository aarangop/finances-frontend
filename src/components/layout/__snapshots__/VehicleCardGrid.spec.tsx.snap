// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCardGrid should render the VehicleCardGrid component 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-c731gg-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-i45qf8-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-u4p24i"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
                  >
                    Car 1
                  </h6>
                  <svg
                    aria-hidden="true"
                    aria-label="CAR"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10umb9k-MuiSvgIcon-root"
                    data-mui-internal-clone-element="true"
                    data-testid="DirectionsCarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16m11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5M5 11l1.5-4.5h11L19 11z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1qfsim1-MuiTypography-root"
                  >
                    Toyota Corolla 2020
                  </p>
                </div>
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-1dfafci-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-e8tykc"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      License Plate
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                    >
                      AAA-111
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      Color
                    </p>
                    <div
                      class="MuiBox-root css-1xhj18k"
                    >
                      <div
                        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1kltci5-MuiChip-root"
                      >
                        <span
                          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                        >
                          WHITE
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-j7qwjs"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                  >
                    Odometer Stand
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                  >
                    12,950
                     km
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
          >
            <div
              class="MuiBox-root css-1dw6sgr"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  href="/vehicles/1"
                  tabindex="0"
                >
                  View
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-i45qf8-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-u4p24i"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
                  >
                    Car 2
                  </h6>
                  <svg
                    aria-hidden="true"
                    aria-label="CAR"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10umb9k-MuiSvgIcon-root"
                    data-mui-internal-clone-element="true"
                    data-testid="DirectionsCarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.92 6.01C18.72 5.42 18.16 5 17.5 5h-11c-.66 0-1.21.42-1.42 1.01L3 12v8c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-1h12v1c0 .55.45 1 1 1h1c.55 0 1-.45 1-1v-8zM6.5 16c-.83 0-1.5-.67-1.5-1.5S5.67 13 6.5 13s1.5.67 1.5 1.5S7.33 16 6.5 16m11 0c-.83 0-1.5-.67-1.5-1.5s.67-1.5 1.5-1.5 1.5.67 1.5 1.5-.67 1.5-1.5 1.5M5 11l1.5-4.5h11L19 11z"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1qfsim1-MuiTypography-root"
                  >
                    BMW X3 2014
                  </p>
                </div>
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-1dfafci-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-e8tykc"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      License Plate
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                    >
                      bbb-111
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      Color
                    </p>
                    <div
                      class="MuiBox-root css-1xhj18k"
                    >
                      <div
                        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1kltci5-MuiChip-root"
                      >
                        <span
                          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                        >
                          RED
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-j7qwjs"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                  >
                    Odometer Stand
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                  >
                    67,909
                     km
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
          >
            <div
              class="MuiBox-root css-1dw6sgr"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  href="/vehicles/2"
                  tabindex="0"
                >
                  View
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 MuiGrid-grid-sm-4 MuiGrid-grid-md-4 css-i45qf8-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-u4p24i"
                >
                  <h6
                    class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
                  >
                    Motorcycle 1
                  </h6>
                  <svg
                    aria-hidden="true"
                    aria-label="MOTORCYCLE"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-10umb9k-MuiSvgIcon-root"
                    data-mui-internal-clone-element="true"
                    data-testid="TwoWheelerIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20 11c-.18 0-.36.03-.53.05L17.41 9H20V6l-3.72 1.86L13.41 5H9v2h3.59l2 2H11l-4 2-2-2H0v2h4c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4l2 2h3l3.49-6.1 1.01 1.01c-.91.73-1.5 1.84-1.5 3.09 0 2.21 1.79 4 4 4s4-1.79 4-4-1.79-4-4-4M4 17c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2m16 0c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2"
                    />
                  </svg>
                </div>
              </span>
              <span
                class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-nrdprl-MuiTypography-root"
              >
                <div
                  class="MuiBox-root css-1xhj18k"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-1qfsim1-MuiTypography-root"
                  >
                    Royal Enfield Himalayan 400 2021
                  </p>
                </div>
              </span>
            </div>
          </div>
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-1dfafci-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-e8tykc"
                >
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      License Plate
                    </p>
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                    >
                      ccc-111
                    </p>
                  </div>
                  <div
                    class="MuiBox-root css-j7qwjs"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                    >
                      Color
                    </p>
                    <div
                      class="MuiBox-root css-1xhj18k"
                    >
                      <div
                        class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1kltci5-MuiChip-root"
                      >
                        <span
                          class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                        >
                          BLACK
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 css-13shxg4-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-j7qwjs"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-fhsa36-MuiTypography-root"
                  >
                    Odometer Stand
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body1 css-bllpwm-MuiTypography-root"
                  >
                    10,560
                     km
                  </p>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-dnrpxu-MuiCardActions-root"
          >
            <div
              class="MuiBox-root css-1dw6sgr"
            >
              <div
                class="MuiBox-root css-i9gxme"
              >
                <a
                  class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-1knaqv7-MuiButtonBase-root-MuiButton-root"
                  href="/vehicles/3"
                  tabindex="0"
                >
                  View
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </a>
              </div>
              <button
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeMedium css-1jo1f9u-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DeleteIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
